### PR TITLE
chore: update typescript to v5

### PR DESCRIPTION
**Pull request by emmenko**

### CODE REVIEW

File: packages/components/inputs/localized-multiline-text-input/src/localized-multiline-text-input.tsx [hunk 1]
```diff
@@ -26,14 +26,7 @@ import TranslationInput from './translation-input';
 import RequiredValueErrorMessage from './required-value-error-message';
 import { warning } from '@commercetools-uikit/utils';
 
-type TState = {
-  [key: string]: string;
-};
-
-type TExpandedTranslationsReducerState = {
-  state?: TState;
-};
```
Comment (emmenko) [inline packages/components/inputs/localized-multiline-text-input/src/localized-multiline-text-input.tsx:35]: This was apparently wrong. Not sure how it got unnoticed until now.

File: packages/components/inputs/select-input/src/select-input.tsx [hunk 2]
```diff
@@ -291,7 +290,7 @@ export type TSelectInputProps = {
   /**
    * Array of options that populate the select menu
    */
-  options?: TOptions;
+  options: TOptions;
```
Comment (emmenko) [inline packages/components/inputs/select-input/src/select-input.tsx:293]: Nothing changed here, the prop is still optional with a default value

File: packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx [hunk 3]
```diff
@@ -319,7 +327,7 @@ const SelectableSearchInput = (props: TSelectableSearchInputProps) => {
     if (props.onSubmit) {
       props.onSubmit({
         text: searchValue,
-        option: selectedOption?.value,
+        option: selectedOption?.value ?? '',
```
Comment (emmenko) [inline packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx:330]: This needs to be double checked. The `option` field is apparently required but the `selectedOption?.value` can be `undefined`.
How should we handle this? Is it fine to return an empty string?

@commercetools/shield-team-fe

**Review by CarlosCortizasCT — APPROVED** [review]
Thanks! 🙌

**Review by kark — APPROVED** [review]
Thank you! 🎉

**Pull request merged**